solc_json_parser/parser.py

```diff
@@ -753,7 +753,7 @@ def __parse_asm_data(self, contract_name, deploy=False) -> Dict[str, Any]:
         asm_data = contract.get('asm').get('.code') if deploy else contract.get('asm').get('.data')
         # deploys = contract.get('asm').get('.code')
         opcodes = contract.get('opcodes').split()
-        source_list = self.get_source_list()
+        source_list = contract.get('asm').get('sourceList') if contract.get('asm').get('sourceList') else self.get_source_list()
```

Comment:
1. Please add some documents here
2. can we move the check `if contract.get('asm').get('sourceList') ...` inside the method `get_source_list`

Reply:
I thought about putting `if contract.get('asm').get('sourceList') ... ` when I was writing the code, and decided not to do it. In function `__parse_asm_data`, you know the contract name already, so you get the source list info inside a contract. But for `get_source_list()`, it is independent of contract name. For the moment, putting the if statement inside `get_source_list()` should work. If you think puting `if contract.get('asm').get('sourceList') ...` is better, I will change it.

Reply:
Sounds reasonable, can you add this information to the comments?

Reply:
of course